server/src/main/java/org/elasticsearch/action/search/ReduceSearchPhaseException.java

```diff
@@ -38,4 +38,15 @@ public ReduceSearchPhaseException(String phaseName, String msg, Throwable cause,
     public ReduceSearchPhaseException(StreamInput in) throws IOException {
         super(in);
     }
+
+    @Override
+    public RestStatus status() {
+        ShardSearchFailure[] shardFailures = super.shardFailures();
+        if (shardFailures.length == 0) {
+            // if no successful shards, the failure can be due to EsRejectedExecutionException during reduce phase
+            // on coordinator node. so get the status from cause instead of returning INTERNAL_SERVER_ERROR blindly
+            return getCause() == null ? RestStatus.INTERNAL_SERVER_ERROR : ExceptionsHelper.status(getCause());
```

Comment:
Looking at where this exception is used, I don't follow when the cause could be null, hence when we would return 500 rather than 503. Can you elaborate?